lib/compress/zstd_compress.c

```diff
@@ -3902,7 +3937,10 @@ size_t ZSTD_compress_generic (ZSTD_CCtx* cctx,
         } else
 #endif
         {
-            CHECK_F( ZSTD_resetCStream_internal(cctx, params, cctx->pledgedSrcSizePlusOne-1) );
+            const void* const prefix = cctx->prefix;
+            size_t const prefixSize = cctx->prefixSize;
+            cctx->prefix = NULL; cctx->prefixSize = 0;   /* single usage */
```

Comment:
Is this comment accurate? It looks like `ZSTD_resetCStream_internal()` will reuse the prefix.